R/seasonal_burden_levels.R

```diff
@@ -115,10 +115,16 @@ seasonal_burden_levels <- function(
     dplyr::slice_max(.data$observation, n = n_peak, with_ties = FALSE, by = "season")
 
   # main level function
-  main_level_fun <- function(seasonal_tsd) {
+  main_level_fun <- function(seasonal_tsd, current_season) {
+
     # Add weights and remove current season to get predictions for this season
-    weighted_seasonal_tsd <- seasonal_tsd |>
-      dplyr::filter(.data$season != max(.data$season)) |>
+    if (current_season %in% unique(seasonal_tsd$season)) {
+      previous_tsd <- seasonal_tsd |>
+        dplyr::filter(.data$season != max(.data$season))
+    } else {
+      previous_tsd <- seasonal_tsd
+    }
+    weighted_seasonal_tsd <- previous_tsd |>
```

Comment:
Would this work instead?

```suggestion
    weighted_seasonal_tsd <- previous_tsd |>
        dplyr::filter(.data$season != current_sesason) |>
```